biojava-structure/src/main/java/org/biojava/nbio/structure/secstruc/SecStrucCalc.java

```diff
@@ -683,7 +683,7 @@ public String printFASTA() {
 	public int hashCode() {
 	    final int prime = 31;
 	    int result = 1;
-	    result = prime * result + Arrays.hashCode(atoms);
+	    result = prime * result + Arrays.hashCode(groups);
```

Comment:
The `equals` compares `SeqStrucInfo` objects are equals per `Group`, whilst above the `hashCode` implementation is based on equality of groups. Instead one would need to build an array of all `SecStrucInfo`s and then compute their hashCode (making sure that `SeqStrucInfo` does implement a hashCode too).